src/webgpu/api/validation/query_set/create.spec.ts

```diff
@@ -0,0 +1,141 @@
+export const description = `
+Tests for validation in createQuerySet.
+`;
+
+import { poptions } from '../../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { ValidationTest } from '../validation_test.js';
+
+export const g = makeTestGroup(ValidationTest);
+
+const kMaxQueryCount = 8192;
+
+// Test with these query counts.
+const kQueryCounts: number[] = [
+  // valid
+  0,
+  kMaxQueryCount,
+  // invalid
+  kMaxQueryCount + 1,
+];
+
+g.test('occlusion')
+  .desc(
+    `
+Tests that create query set with the type of occlusion.
+- control cases:
+  - count <= kMaxQueryCount
+  - pipelineStatistics is undefined or empty
+- invalid cases:
+  - count > kMaxQueryCount
+  - pipelineStatistics is set with pipeline statistic names
+  `
+  )
+  .params([
+    ...poptions('count', kQueryCounts),
+    ...poptions('pipelineStatistics', [
+      // valid
+      undefined,
+      [],
+      // invalid
+      ['clipper-invocations'],
+    ]),
+  ])
+  .fn(t => {
+    const { count = 1, pipelineStatistics } = t.params as {
+      count?: number;
+      pipelineStatistics?: GPUPipelineStatisticName[];
+    };
+
+    t.expectValidationError(() => {
+      t.device.createQuerySet({ type: 'occlusion', count, pipelineStatistics });
+    }, count > kMaxQueryCount || (pipelineStatistics !== undefined && pipelineStatistics.length > 0));
+  });
+
+g.test('pipeline_statistics')
+  .desc(
+    `
+Tests that create query set with the type of pipeline-statistics.
+- control cases:
+  - count <= kMaxQueryCount
+  - pipelineStatistics is set with pipeline statistic names
+- invalid cases:
+  - count > kMaxQueryCount
+  - pipelineStatistics is undefined, empty or duplicated
+  `
+  )
+  .params([
+    ...poptions('count', kQueryCounts),
+    ...poptions('pipelineStatistics', [
+      // valid
+      ['clipper-invocations'],
+      // invalid
+      undefined,
+      [],
+      ['clipper-invocations', 'clipper-invocations'],
+    ]),
+  ])
+  .fn(async t => {
+    await t.selectDeviceOrSkipTestCase({
```

Comment:
Can we also test that creating a pipeline stastistics query / timestamp query set is invalid if we don't have the extension?

Reply:
Nevermind, that's done in another PR.